daml-lf/transaction/src/main/scala/com/digitalasset/daml/lf/transaction/Transaction.scala

```diff
@@ -364,13 +364,30 @@ sealed abstract class HasTxNodes[Nid, +Cid] {
     globalState
   }
 
-  final def localContracts[Cid2 >: Cid]: Map[Cid2, Nid] =
-    fold(Map.empty[Cid2, Nid]) {
+  final def localContracts[Cid2 >: Cid]: Map[Cid2, (Nid, Node.NodeCreate[Cid2])] =
+    fold(Map.empty[Cid2, (Nid, Node.NodeCreate[Cid2])]) {
       case (acc, (nid, create: Node.NodeCreate[Cid])) =>
-        acc.updated(create.coid, nid)
+        acc.updated(create.coid, nid -> create)
       case (acc, _) => acc
     }
 
+  /** Returns the IDs of all the consumed contracts.
+    *  This includes transient contracts but it does not include contracts
+    *  consumed in rollback nodes.
+    */
+  final def consumedContracts[Cid2 >: Cid]: Set[Cid2] =
+    foldInExecutionOrder(Set.empty[Cid2])(
+      exerciseBegin = (acc, _, exe) => {
+        val acc0 = if (exe.consuming) { acc + exe.targetCoid }
```

Comment:
You might want to use braces around all the if branches -- the indentation looks confusing otherwise. You could even simplify it by returning directly from the branches:
```scala
  if (exe.consuming) {
    (acc + exe.targetCoid, true)
  } else {
    (acc, true)
  }
```

Reply:
413ba40cbbdb20b51d31575ccadbd3551e3f323b